opendatahub-io/data-science-pipelines-operator
# Updated handling of dspa tls config

## The issue resolved by this Pull Request:
Resolves [RHOAIENG-4384](https://issues.redhat.com/browse/RHOAIENG-4384)

## Description of your changes:
Removed the string template for default value of DbExtraParams and created a struct to handle the tls config.

## Testing instructions

- Use dspa_simple.yaml file to deploy dspa and  create a pipeline run to verify if it runs properly.
- Use the following yaml to deploy  dspa and create a pipeline run to verify if it runs properly
```
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: sample
spec:
  dspVersion: v2
  apiServer:
    deploy: true    
    enableSamplePipeline: true
  objectStorage:
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
  database:
    customExtraParams: '{"tls":"false"}'
    externalDB:
      host: mariadb.test-mariadb.svc.cluster.local
      port: "3306"
      username: mlpipeline
      pipelineDBName: mlpipeline
      passwordSecret:
        name: ds-pipeline-db-test
        key: password
  mlpipelineUI:
    image: quay.io/opendatahub/ds-pipelines-frontend:latest

```

## Checklist
